cli/report/cmd.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package report
+
+import (
+	"errors"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var flags struct {
+	queryPath    string
+	reportPath   string
+	reportFormat string
+	bucketName   string
+	caiDirName   string
+	listReports  bool
+}
+
+func init() {
+	viper.AutomaticEnv()
+
+	Cmd.Flags().StringVar(&flags.queryPath, "query-path", "", "Path to directory containing inventory queries")
+	Cmd.MarkFlagRequired("query-path")
+
+	Cmd.Flags().StringVar(&flags.reportPath, "report-path", "", "Path to directory to contain report outputs")
+
+	Cmd.Flags().StringVar(&flags.bucketName, "bucket", "", "GCS bucket name for storing inventory (conflicts with --local-cai-path)")
+	Cmd.Flags().StringVar(&flags.caiDirName, "local-cai-path", "", "Local directory path for storing inventory (conflicts with --bucket)")
+
+	Cmd.Flags().StringVar(&flags.reportFormat, "format", "", "Format of inventory report outputs, can be json or csv, default is csv")
```

Comment:
We might want to use Viper to manage these flags, as it includes default-setting functionality: https://github.com/spf13/viper#establishing-defaults

Reply:
done in https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/315/commits/8d47e44125e9ad0d4c53bf2b7f88642ccdd9b286